core/primitives/src/sharding.rs

```diff
@@ -194,11 +196,36 @@ impl ShardChunkHeaderV3 {
         tx_root: CryptoHash,
         prev_validator_proposals: Vec<ValidatorStake>,
         congestion_info: Option<CongestionInfo>,
+        bandwidth_requests: Option<BandwidthRequests>,
         signer: &ValidatorSigner,
     ) -> Self {
-        let inner = if let Some(congestion_info) = congestion_info {
+        let Some(congestion_info) = congestion_info else {
+            // Old inner without congestion info
+            let inner = ShardChunkHeaderInner::V2(ShardChunkHeaderInnerV2 {
+                prev_block_hash,
+                prev_state_root,
+                prev_outcome_root,
+                encoded_merkle_root,
+                encoded_length,
+                height_created: height,
+                shard_id,
+                prev_gas_used,
+                gas_limit,
+                prev_balance_burnt,
+                prev_outgoing_receipts_root,
+                tx_root,
+                prev_validator_proposals,
+            });
+            return Self::from_inner(inner, signer);
+        };
+        // `congestion_info`` can only be `Some` when congestion control is enabled.
+        assert!(ProtocolFeature::CongestionControl.enabled(protocol_version));
+
+        let bandwidth_requests = bandwidth_requests.unwrap_or_else(BandwidthRequests::empty);
+
+        let inner = if ProtocolFeature::BandwidthScheduler.enabled(protocol_version) {
```

Comment:
nit: Maybe follow the same pattern as for congestion control. I think it would be nice to keep the code in this method in chronological order (V2, V3, V4). 
```
let Some(congestion_info) = congestion_info else {
  return V2(...); 
}
assert!(ProtocolFeature::CongestionControl.enabled(protocol_version));

let Some(bandwidth_requests) = bandwidth_requests else {
  return V3(...); 
}
assert!(ProtocolFeature::BandwidthScheduler.enabled(protocol_version));

return V4(...);
```